packages/BRC721/src/brc721Wallet.ts

```diff
@@ -0,0 +1,41 @@
+import { BRC721 } from './brc721'
+
+interface IBRC721Wallet {
+  balanceOf(owner: string): Promise<number>
+  ownerOf(tokenId: string): Promise<string[]>
+  transferTo(to: string, tokenId: string)
+  mint(to: string, name?: string, symbol?: string): Promise<BRC721>
+}
+
+export class BRC721Wallet implements IBRC721Wallet {
+  computer: any
+  contract: any
+
+  constructor(computer: any, contract: any = BRC721) {
+    this.computer = computer
+    this.contract = contract
+  }
+
+  async balanceOf(publicKey: string): Promise<number> {
+    const revs = await this.computer.queryRevs({ publicKey, contract: this.contract })
+    const nfts: any[] = await Promise.all(revs.map((rev) => this.computer.sync(rev)))
```

Comment:
Because NFTs cannot be deleted in this contract we do not need to sync and can call ``this.contract.balanceOf(revs)``.